test/api/bucket/api_test.go

```diff
@@ -0,0 +1,84 @@
+package bucket
+
+import (
+	"testing"
+	"time"
+
+	"github.com/konveyor/tackle2-hub/api"
+	"github.com/konveyor/tackle2-hub/binding"
+	"github.com/konveyor/tackle2-hub/test/assert"
+)
+
+func TestBucketCRUD(t *testing.T) {
+	for _, sample := range Samples {
+		t.Run(sample.Path, func(t *testing.T) {
+
+			expectedBucket := api.Bucket{
+				Path: sample.Path,
+			}
+
+			// Create a new bucket.
+			assert.Must(t, Bucket.Create(&expectedBucket))
+
+			// Get all the buckets.
+			_, err := Bucket.List()
+			if err != nil {
+				t.Errorf(err.Error())
+			}
+
+			// Inject Expected Buckets's ID into the BucketRoot.
+			pathForBucket := binding.Path(api.BucketRoot).Inject(binding.Params{api.ID: expectedBucket.ID})
+
+			// Get specific buckets
+			gotBucket := api.Bucket{}
+			err = Client.Get(pathForBucket, &gotBucket)
+			if err != nil {
+				t.Errorf(err.Error())
+			}
+
+			// Compare bucket Paths
+			if gotBucket.Path != expectedBucket.Path {
+				t.Errorf("Difference in Path between the buckets %v and %v", gotBucket.Path, expectedBucket.Path)
+			}
+
+			// Delete bucket.
+			assert.Must(t, Bucket.Delete(pathForBucket))
+		})
+	}
+}
+
+// func TestBucketFile(t *testing.T) {
+// 	for _, sample := range Samples {
+// 		t.Run("Bucket File Test", func(t *testing.T) {
+// 			err := Client.BucketPut(sample.Path, api.BucketContentRoot)
+// 			if err != nil {
+// 				t.Errorf(err.Error())
+// 			}
+// 		})
+// 	}
+// }
+
+func TestBucketReaper(t *testing.T) {
```

Comment:
This seems outside the scope of the API test.

Reply:
Hi Jeff Thanks for the review. Actually me and Marek discussed about the Expiration tag in Bucket and wanted to test indirect deletion of the bucket upon Expiration. So we had decided to create a sample bucket and check its expiration

Reply:
Will do that!

Reply:
https://github.com/konveyor/tackle2-hub/issues/446